contextid_test.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func getIdFromRequest(t *testing.T, server *httptest.Server) uint {
+	res, err := http.Get(server.URL)
+	assert.NoError(t, err)
+	b, err := ioutil.ReadAll(res.Body)
+	assert.NoError(t, err)
+	id, err := strconv.ParseUint(string(b), 10, 64)
+	assert.NoError(t, err)
+	return uint(id)
+}
+
+func TestContextId(t *testing.T) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		id := r.Context().Value("id")
+		_, err := w.Write([]byte(strconv.FormatUint(uint64(id.(uint)), 10)))
```

Comment:
more likely (whatever that means) that the type assertion panics than that w.Write errors I reckon.

Reply:
I think you might be saying that I should test explicitly for the type of the id, and I agree. I will add that test. As to handling the `Write` error, that was to silence a lint warning (and since I wrote the code, I handled the error). Is there a common way in tests to differentiate between assertions that validate the results of the test and assertions that just make sure the test works as expected? This is the latter so I don't want to ignore the error - may as well fail the test and understand why the error occurred, but it might be nice for the developer to know that the error is in the test case and not the unit under test.

Reply:
I've split out the type assertion and asserted that it is ok. So no panic now, but it is tested that the type is correct. I didn't make a new test for this. New code coming with the next push.

Reply:
> strong guarantee about what's coming out

Well, only if the code is correct. And the purpose of these tests is to confirm that. I think an assertion is still the correct form of detection of incorrect code.

Reply:
Fixed in 9c865583b25d7fd8d8c4315c2ccd2cbe7ae90155. Will rebase-squash prior to merging.